### PR TITLE
[ARCTIC-1640] Splitting different thrift processors into different thread pools

### LINKED ISSUE #1640: [Improvement]: Splitting different thrift processors into different thread pools.
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/arctic/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Splitting different thrift processors into different thread pools.

### How should we improve?

Using different startup parameters to configure different thread pools when starting different Thrift processors.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by hameizi**
## Why are the changes needed?

resolve #1640 

## Brief change log

Start separate Thrift servers for the table management service and optimizing service (using different server port) and configure different thread pools for each.

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (no)
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

File: ams/server/src/main/java/com/netease/arctic/server/ArcticServiceContainer.java [hunk 1]
```diff
@@ -158,11 +163,18 @@ private void initConfig() throws IOException {
...
+    }, "Thrift-table-management-server-thread");
     thread.setDaemon(true);
     thread.start();
+    LOG.info("Table management thrift services have been started");
+
+    Thread optimizingThread = new Thread(() -> {
+      optimizingServer.serve();
+    }, "Thrift-optimizing-server-thread");
+    optimizingThread.setDaemon(true);
+    optimizingThread.start();
```
Comment (majin1102) [inline ams/server/src/main/java/com/netease/arctic/server/ArcticServiceContainer.java:176]: Is optimizingThread a conflict concept?

File: ams/server/src/main/java/com/netease/arctic/server/ArcticServiceContainer.java [hunk 2]
```diff
@@ -183,37 +193,45 @@ private void initThriftService() throws TTransportException {
...
+        createThriftServer(optimizingProcessor, Constants.THRIFT_OPTIMIZING_SERVICE_NAME, serverTransport,
+            maxMessageSize, Executors.newCachedThreadPool(), selectorThreads, queueSizePerSelector);
+  }
 
-    TNonblockingServerSocket serverTransport = getServerSocket(bindHost, port);
+  private TServer createThriftServer(
+      TProcessor processor, String processorName,
+      TNonblockingServerSocket serverTransport, long maxMessageSize,
+      ExecutorService executorService, int selectorThreads, int queueSizePerSelector) {
+    LOG.info(String.format("Initializing thrift server: %s", processorName));
```
Comment (zhoujinsong) [inline ams/server/src/main/java/com/netease/arctic/server/ArcticServiceContainer.java:217]:
```suggestion
    LOG.info("Initializing thrift server:{}", processorName));
```

File: ams/server/src/main/java/com/netease/arctic/server/ArcticServiceContainer.java [hunk 3]
```diff
@@ -183,37 +193,45 @@ private void initThriftService() throws TTransportException {
...
-        .workerThreads(workerThreads)
+        .executorService(executorService)
         .selectorThreads(selectorThreads)
         .acceptQueueSizePerThread(queueSizePerSelector);
-    server = new TThreadedSelectorServer(args);
-    LOG.info("Initialized the thrift server on port [" + port + "]...");
-    LOG.info("Options.thriftWorkerThreads = " + workerThreads);
-    LOG.info("Options.thriftSelectorThreads = " + selectorThreads);
-    LOG.info("Options.queueSizePerSelector = " + queueSizePerSelector);
+    LOG.info(String.format("The number of selector threads for the %s thrift server is: %d", processorName,
```
Comment (zhoujinsong) [inline ams/server/src/main/java/com/netease/arctic/server/ArcticServiceContainer.java:230]: use `Log.info(String format, Object... arguments)` instaead.

File: ams/server/src/main/java/com/netease/arctic/server/ArcticServiceContainer.java [hunk 4]
```diff
@@ -183,37 +193,45 @@ private void initThriftService() throws TTransportException {
...
-    inputProtoFactory = new TBinaryProtocol.Factory(true, true, maxMessageSize, maxMessageSize);
-
-    ArcticTableMetastore.Processor<ArcticTableMetastore.Iface> tableMetastoreProcessor =
+    ArcticTableMetastore.Processor<ArcticTableMetastore.Iface> tableManagementProcessor =
         new ArcticTableMetastore.Processor<>(ThriftServiceProxy.createProxy(ArcticTableMetastore.Iface.class,
             new TableManagementService(tableService), ArcticRuntimeException::normalizeCompatibly));
-    processor.registerProcessor(Constants.THRIFT_TABLE_SERVICE_NAME, tableMetastoreProcessor);
+    tableManagementServer =
+        createThriftServer(tableManagementProcessor, Constants.THRIFT_TABLE_SERVICE_NAME, serverTransport,
+            maxMessageSize, Executors.newFixedThreadPool(workerThreads), selectorThreads, queueSizePerSelector);
```
Comment (zhoujinsong) [inline ams/server/src/main/java/com/netease/arctic/server/ArcticServiceContainer.java:203]: use `Executors.newFixedThreadPool(int nThreads, ThreadFactory threadFactory)` to set the thread name pattern for executors.

File: ams/server/src/main/java/com/netease/arctic/server/ArcticServiceContainer.java [hunk 5]
```diff
@@ -183,37 +193,45 @@ private void initThriftService() throws TTransportException {
...
 
-    OptimizingService.Processor<OptimizingService.Iface> optimizerManagerProcessor =
+    OptimizingService.Processor<OptimizingService.Iface> optimizingProcessor =
         new OptimizingService.Processor<>(ThriftServiceProxy.createProxy(OptimizingService.Iface.class,
             optimizingService, ArcticRuntimeException::normalize));
-    processor.registerProcessor(Constants.THRIFT_OPTIMIZING_SERVICE_NAME, optimizerManagerProcessor);
+    optimizingServiceServer =
+        createThriftServer(optimizingProcessor, Constants.THRIFT_OPTIMIZING_SERVICE_NAME, serverTransport,
+            maxMessageSize, Executors.newCachedThreadPool(), selectorThreads, queueSizePerSelector);
```
Comment (zhoujinsong) [inline ams/server/src/main/java/com/netease/arctic/server/ArcticServiceContainer.java:210]: use `Executors.newCachedThreadPool(ThreadFactory threadFactory)` to set the thread name pattern for executors.

**Review by majin1102 — APPROVED** [review]
LGTM

**Pull request merged**